### PR TITLE
Reduce the amount of retries for too early events

### PR DESCRIPTION
**Pull request by karlb**
Closes https://github.com/raiden-network/raiden-services/issues/1153.

I also cherry picked d9f200dac55ed00ff19348783ebedf2a20ebe50d from master, along with 9150691.

### CODE REVIEW

File: src/monitoring_service/service.py [hunk 1]
```diff
@@ -197,8 +191,22 @@ def _trigger_scheduled_events(self) -> None:
...
-            self.context.database.remove_scheduled_event(scheduled_event)
+            try:
+                handle_event(event, self.context)
+            except TransactionTooEarlyException:
+                log.debug(
+                    "Event executed too early. "
+                    "Retry later and don't try any other scheduled events right now.",
+                    handled_event=event,
+                )
+                # When the scheduled event with the lowest timestamp fails with
```
Comment (weilbith) [inline src/monitoring_service/service.py:206]: From the code in the database module I'm not sure: is the event list actually sorted by timestamps?
In any case I would probably help to also express this somewhere. At least in the docs of the database function or even in the function name itself.

**Review by weilbith — CHANGES_REQUESTED** [review]
I like the basic solution. But as we both discussed, I think we need a new queue only for transactions which follows this approach. The problem is that the current priority queue of scheduled events we are working on is also used for many other events. Their handlers do not execute a transaction on the blockchain. I'm really not sure if it is good or possible to block all of these other events as they in-fact define the whole reactive functionality of the service. All the following events get blocked without a transaction in their handler:
- `UpdatedHeadBlockEvent`
- `ReceiveTokenNetworkCreatedEvent`
- `ReceiveChannelOpenedEvent`
- `ReceiveMonitoringNewBalanceProofEvent`
- `ReceiveChannelClosedEvent`
- `ReceiveChannelSettledEvent`
- `ReceiveMonitoringRewardClaimedEvent`

Especially the `UpdatedHeadBlockEvent` is quite important as it sets the internal blockchain state which is used to fetch new blocks if I'm not mistaken.
So are you really sure we can get away with blocking the main event queue here?

File: src/monitoring_service/service.py (hunk 1 above)
Comment (karlb) [reply src/monitoring_service/service.py:206]: Good catch! We really need to sort explicitly now. I'll push a commit for that.

Comment (karlb) [reply src/monitoring_service/service.py:206]: Good catch! We really need to sort explicitly now. I'll push a commit for that.

Comment (karlb) [reply src/monitoring_service/service.py:206]: Good catch! We really need to sort explicitly now. I'll push a commit for that.

Comment (weilbith) [reply src/monitoring_service/service.py:206]: Got resolved with [e66c720](https://github.com/raiden-network/raiden-services/pull/1158/commits/e66c7204fb65678bcff63a51a35a789f1ca10bea) 👍🏾

**Review by weilbith — APPROVED** [review]
I'm positive that this does the job now. 👍🏾 
Let's see how it looks in practice.

**Pull request merged**